MetabolicAtlas/MetabolicAtlas
# fix: interaction partners viewer height

**Related issue(s) and PR(s)**  
This PR closes #1176 and #1207.



**Type of change**  

 
**List of changes made**  

1. Set fixed viewer height after page load.
2. Fix mobile viewer height after page load. (This is and unreported bug, where viewer height is `0`.)
3. Add handler to adjust viewer height after window resize.

**Testing**  

- Relative urls that can be reused both for production and local testing
- Instructions on how to test

1. Visit an interaction partners page. For example: `/explore/Human-GEM/interaction-partners/ENSG00000069667`.
2. Click any node, which should slightly expand the sidebar height.
3. Verify that the labels do not shift. (For issue: #1176)
4. Upload data for `metabolites` in Data Overlay and click the "Add overlay" button.
5. Verify that the network viewer does not increase in height. (For issue: #1207)
6. Resize the window to mobile width.
7. Verify that the height adjusts accordingly.
8. Refresh the page while still on mobile width.
9. Verify that the viewer loads and is visible. (Item 2 under "List of changes made".)

**Further comments**  

Since the shifting in labels is related to the expandable network viewer height, this PR also targets #1207.

**Definition of Done checklist**
